cubed/runtime/executors/dask.py

```diff
@@ -37,4 +42,15 @@ def execute_dag(
                     )
                     stage_delayed_funcs = [delayed_func]
 
-                dask.compute(*stage_delayed_funcs, **compute_kwargs)
+                def dask_callback(key, result, dsk, state, id):
+                    _, stats = result
+                    stats["array_name"] = name
+                    handle_callbacks(callbacks, stats)
+
+                if callbacks is not None:
+                    cm = Callback(posttask=dask_callback)
```

Comment:
I just realised that this only works for local Dask schedulers, not for distributed. So probably best to drop this and exclude the callbacks test for Dask.